apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java

```diff
@@ -1,4 +1,59 @@
 package ch.heig.pdg.backend.services;
 
-public class InventoryService {
+import ch.heig.pdg.backend.dto.InventoryDTO;
+import ch.heig.pdg.backend.dto.mapping.InventoryMapper;
+import ch.heig.pdg.backend.entities.Inventory;
+import ch.heig.pdg.backend.repositories.InventoryRepository;
+import ch.heig.pdg.backend.utils.HugoSearchFilter;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class InventoryService extends AbstractService {
+    private final InventoryMapper inventoryMapper;
+
+    protected InventoryService(InventoryRepository inventoryRepository, InventoryMapper inventoryMapper) {
+        super(inventoryRepository);
+        this.inventoryMapper = inventoryMapper;
+    }
+
+    public List<InventoryDTO> getInventories(HugoSearchFilter<Inventory> filter) {
+        return this.inventoryRepository
+                .findByFilter(filter)
+                .stream()
+                .map(inventory -> (InventoryDTO) this.inventoryMapper.getDTO(inventory))
+                .collect(Collectors.toList());
+    }
+
+    public InventoryDTO getInventory(Integer id) {
+        return (InventoryDTO) this.inventoryMapper.getDTO(
+                this.getEntityIfExists(id, this.inventoryRepository)
+        );
+    }
+
+    public InventoryDTO updateInventory(Integer id, InventoryDTO inventoryDTO) {
+        Inventory inventory = this.getEntityIfExists(id, this.inventoryRepository);
+
+        return (InventoryDTO) this.inventoryMapper.getDTO(
+                this.inventoryRepository.save(
+                        this.inventoryMapper.updateFromDTO(inventory, inventoryDTO)
+                )
+        );
+    }
+
+    public InventoryDTO addInventory(InventoryDTO inventoryDTO) {
+        Inventory inventory = this.inventoryMapper.createFromDTO(inventoryDTO);
+
+        return (InventoryDTO) this.inventoryMapper.getDTO(
+                this.inventoryRepository.save(inventory)
```

Comment:
Do you happen to have a 5inch vertical monitor and lines can't be longer than 20 characters ?
It's Java not python !